skimage/feature/fisher_vector.py

```diff
@@ -0,0 +1,216 @@
+by the user, namely:
+
+learn_gmm: Used to estimate the GMM for all vectors/descriptors computed for
+           all examples in the dataset (e.g. estimated using all the SIFT
+           vectors computed for all images in the dataset, or at least a subset
+           of this).
+
+compute_fisher_vector: Used to compute the Fisher vector representation for a
+                       single set of descriptors/vector (e.g. the SIFT
+                       descriptors for a single image in your dataset, or
+                       perhaps a test image).
+
+Reference: Perronnin, F. and Dance, C. Fisher kernels on Visual Vocabularies
+           for Image Categorization, IEEE Conference on Computer Vision and
+           Pattern Recognition, 2007
+
+Origin Author: Dan Oneata (Author of the original implementation for the Fisher
+vector computation using scikit-learn and NumPy. Subsequently ported to
+scikit-image (here) by other authors.)
+"""
+import numpy as np
+
+
+class FisherVectorException(Exception):
+    pass
+
+
+class DescriptorException(FisherVectorException):
+    pass
+
+
+def learn_gmm(descriptors, num_modes, **kwargs):
+    """Function to estimate a GMM given a set of descriptors and number of
+    modes (i.e. Gaussians). This function is essentially a wrapper around
+    the scikit-learn implementation of GMM, namely the
+    sklearn.mixture.GaussianMixture class.
+
+    Due to the nature of the Fisher vector, the only enforce parameter of the
+    underlying scikit-learn class is the covariance_type, which must be 'diag'.
+
+    Parameters
+    -----------
+    descriptors : np.ndarray or list
+        List of NumPy arrays, or a single NumPy array, of the descriptors
+        to use to estimate the GMM. Reason a list of NumPy arrays is
+        permissible is because often when using a Fisher vector encoding,
+        descriptors/vectors are computed separately for each sample/image in
+        the dataset, such as SIFT vectors for each image.
+    num_modes : int
+        The number of modes/Gaussians to estimate during the GMM estimate.
+    **kwargs : dict
+        Keyword arguments that can be passed into the underlying scikit-learn
+        GaussianMixture class.
+
+    Returns
+    -------
+    gmm : sklearn.mixture.GaussianMixture
+        The estimated GMM object, which contains the necessary parameters
+        needed to compute the Fisher vector.
+
+    References
+    -----------
+    .. [1] https://scikit-learn.org/stable/modules/generated/sklearn.mixture.GaussianMixture.html
+
+    Examples
+    --------
+    >>> from skimage.feature import fisher_vector
+    >>> sift_for_images = [np.random.random((10, 128)) for _ in range(10)]
+    >>> num_modes = 16
+    >>> # Estimate 16-mode GMM with these synthetic SIFT vectors
+    >>> gmm = fisher_vector.learn_gmm(sift_for_images, num_modes)
+    """
+
+    try:
+        from sklearn.mixture import GaussianMixture
+    except ImportError:
+        print(
```

Comment:
```suggestion
        raise ImportError(
```
This should still raise an error. We can just skip the tests for the CI jobs where optional dependencies are not installed. See [pytest.importorskip](https://docs.pytest.org/en/stable/reference.html#pytest-importorskip).

Reply:
Added

Reply:
Done